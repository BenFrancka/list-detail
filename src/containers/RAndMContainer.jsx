import React, { Component } from 'react';
import { fetchCharacters } from '../services/RAndMApi';
import CharacterList from '../components/CharacterList';

export default class RAndMContainer extends Component {
  state = {
    loading: true,
    characters: [],
  };

  async componentDidMount() {
    const characters = await fetchCharacters();
    this.setState({ characters, loading: false });
  }

  render() {
    const { characters, loading } = this.state;
    if(loading) {
      return (
        <img
          src="https://thumbs.gfycat.com/SoftEarnestGyrfalcon-max-1mb.gif"
          alt="loading spinner"
        />
      );
    }
    return <CharacterList characters={characters} />;
  }
}
