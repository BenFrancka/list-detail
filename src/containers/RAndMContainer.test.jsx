import React from 'react';
import { render, screen } from '@testing-library/react';
import RAndMContainer from './RAndMContainer';

describe('RAndMContainer', () => {
  it('renders a list of Rick and Morty Characters to home page', async () => {
    render(<RAndMContainer />);

    screen.getByAltText('loading spinner');

    const ul = await screen.findByRole('list', {
      name: 'rick and morty characters',
    });

    expect(ul).toMatchSnapshot();
  });
});
