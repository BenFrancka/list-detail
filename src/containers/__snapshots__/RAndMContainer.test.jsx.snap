// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RAndMContainer renders a list of Rick and Morty Characters to home page 1`] = `
<ul
  aria-label="rick and morty characters"
>
  <a
    href="/1"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
      <h2>
        Rick Sanchez
      </h2>
    </li>
  </a>
  <a
    href="/2"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
      />
      <h2>
        Morty Smith
      </h2>
    </li>
  </a>
  <a
    href="/3"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
      />
      <h2>
        Summer Smith
      </h2>
    </li>
  </a>
  <a
    href="/4"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/4.jpeg"
      />
      <h2>
        Beth Smith
      </h2>
    </li>
  </a>
  <a
    href="/5"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/5.jpeg"
      />
      <h2>
        Jerry Smith
      </h2>
    </li>
  </a>
  <a
    href="/6"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/6.jpeg"
      />
      <h2>
        Abadango Cluster Princess
      </h2>
    </li>
  </a>
  <a
    href="/7"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/7.jpeg"
      />
      <h2>
        Abradolf Lincler
      </h2>
    </li>
  </a>
  <a
    href="/8"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/8.jpeg"
      />
      <h2>
        Adjudicator Rick
      </h2>
    </li>
  </a>
  <a
    href="/9"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/9.jpeg"
      />
      <h2>
        Agency Director
      </h2>
    </li>
  </a>
  <a
    href="/10"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/10.jpeg"
      />
      <h2>
        Alan Rails
      </h2>
    </li>
  </a>
  <a
    href="/11"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/11.jpeg"
      />
      <h2>
        Albert Einstein
      </h2>
    </li>
  </a>
  <a
    href="/12"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/12.jpeg"
      />
      <h2>
        Alexander
      </h2>
    </li>
  </a>
  <a
    href="/13"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/13.jpeg"
      />
      <h2>
        Alien Googah
      </h2>
    </li>
  </a>
  <a
    href="/14"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/14.jpeg"
      />
      <h2>
        Alien Morty
      </h2>
    </li>
  </a>
  <a
    href="/15"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/15.jpeg"
      />
      <h2>
        Alien Rick
      </h2>
    </li>
  </a>
  <a
    href="/16"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/16.jpeg"
      />
      <h2>
        Amish Cyborg
      </h2>
    </li>
  </a>
  <a
    href="/17"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/17.jpeg"
      />
      <h2>
        Annie
      </h2>
    </li>
  </a>
  <a
    href="/18"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/18.jpeg"
      />
      <h2>
        Antenna Morty
      </h2>
    </li>
  </a>
  <a
    href="/19"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/19.jpeg"
      />
      <h2>
        Antenna Rick
      </h2>
    </li>
  </a>
  <a
    href="/20"
  >
    <li>
      <img
        src="https://rickandmortyapi.com/api/character/avatar/20.jpeg"
      />
      <h2>
        Ants in my Eyes Johnson
      </h2>
    </li>
  </a>
</ul>
`;
